api/src/main/java/org/apache/iceberg/transforms/Days.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.transforms;
+
+import java.io.ObjectStreamException;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+
+public class Days<T> extends TimeTransform<T> {
+  private static final Days<?> INSTANCE = new Days<>();
+
+  @SuppressWarnings("unchecked")
+  static <T> Days<T> get() {
+    return (Days<T>) INSTANCE;
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  protected Transform<T, Integer> toEnum(Type type) {
+    switch (type.typeId()) {
+      case DATE:
+        return (Transform<T, Integer>) Dates.DAY;
+      case TIMESTAMP:
+        return (Transform<T, Integer>) Timestamps.DAY;
+    }
+    throw new IllegalArgumentException("Unsupported type: " + type);
```

Comment:
nit: It might be a personal thing, but I find having a default branch and throwing an exception from there a bit more obvious.

Reply:
Fixed.